### DISCUSSION TITLE
Render option for last-n animations

### DISCUSSION THREAD

**Discussion by mscuthbert** [ManimCommunity/manim#2814] [Suggestions and Proposals]
## Description of proposed feature

In rendering manim videos, I often want to render just the last few animations.  I'll use the flag `-n 35` and then when I've added a few more animations, I'll update to `-n 40`, etc.

It would be helpful to have a flag like `-n -5` or `--last_n_frames 5` which would only render the last few frames of the video; saving time when debugging with early returns, etc.

## How can the new feature be used?

as described above -- to save time when developing, especially in CairoRenderer where interactive debug isn't available.

## Additional comments

I'm still a manim novice -- finishing my first longish video (medieval music animations!) so I've only just gotten a grasp of how the Scene and renderers interact with each other, but my sense is that the difficulty of this feature is that when `render()` is called and `construct()` is run, the system has no idea how many `play()` calls will be run, so counting backwards isn't easy.

Two implementation paths come to mind, both involving running the scene twice.

1. use the existing save_last_snapshot mechanism to count the number of plays and use that information to set `-n start-end` appropriately and run again.
2. create a new Renderer class `NullRenderer` whose `.play()` function does nothing except `update_skipping_status`.  Creating a `BaseRenderer` that Cairo and OpenGL (and NullRenderer) both inherit from would probably save some code-dup anyhow.

The problem with both of these approaches is if the `construct()` call changes properties on `self` while running and does not reset them they will be incorrect the second run:

```
class MyScene(Scene):
    def __init__(self, ...):
          ...
          self.start_height = 10

    def construct(self):
          ...
          self.play(obj.animate.shift(self.start_height))
          self.start_height += 5
          self.play(obj.animate.shift(self.start_height))
```

This could be a documented limitation of this method or doing something like creating a deepcopy of the Scene object before running construct() could work.  It still would have issues though if there were side effects like reading/writing state from disk, but that is quite rare.

If the community supported this, I'd be willing to implement it myself.

**Comment by tameTNT** [comment]
The implementation of this feature (specifically `--last_n_frames`) will likely have some overlap with #2589

**Comment by ad-chaos** [comment]
>  I often want to render just the last few animations

Have you taken a look at [`Sections`](https://docs.manim.community/en/stable/tutorials/a_deeper_look.html?highlight=Sections#sections) ?
